src/devices/vdml_adi.c

```diff
@@ -10,371 +10,110 @@
-	validate_type(device, port, E_ADI_ANALOG_IN);
-	uint32_t total = 0;
-	for (uint32_t i = 0; i < 512; i++) {
-		total += vexDeviceAdiValueGet(device->device_info, port);
-		task_delay(1); // TODO: If smart ports (and the ADI) only update every 10ms, this really only reads 56 samples, maybe change to a 10ms
-	}
-	adi_data_s_t* const adi_data = &((adi_data_s_t*)(device->pad))[port];
-	adi_data->analog_data.calib = (int32_t)((total + 16) >> 5);
-	return_port(INTERNAL_ADI_PORT, (int32_t)((total + 256) >> 9));
+	return ext_adi_analog_calibrate(port, INTERNAL_ADI_PORT);
 }
 
 int32_t adi_analog_read(uint8_t port) {
-	transform_adi_port(port);
-	claim_port_i(INTERNAL_ADI_PORT, E_DEVICE_ADI);
-	validate_type(device, port, E_ADI_ANALOG_IN);
-	int32_t rtn = vexDeviceAdiValueGet(device->device_info, port);
-	return_port(INTERNAL_ADI_PORT, rtn);
+	return ext_adi_analog_read(port, INTERNAL_ADI_PORT);
 }
 
 int32_t adi_analog_read_calibrated(uint8_t port) {
-	transform_adi_port(port);
-	claim_port_i(INTERNAL_ADI_PORT, E_DEVICE_ADI);
-	validate_type(device, port, E_ADI_ANALOG_IN);
-	adi_data_s_t* const adi_data = &((adi_data_s_t*)(device->pad))[port];
-	int32_t rtn = (vexDeviceAdiValueGet(device->device_info, port) - (adi_data->analog_data.calib >> 4));
-	return_port(INTERNAL_ADI_PORT, rtn);
+	return ext_adi_analog_read_calibrated(port, INTERNAL_ADI_PORT);
 }
 
 int32_t adi_analog_read_calibrated_HR(uint8_t port) {
-	transform_adi_port(port);
-	claim_port_i(INTERNAL_ADI_PORT, E_DEVICE_ADI);
-	validate_type(device, port, E_ADI_ANALOG_IN);
-	adi_data_s_t* const adi_data = &((adi_data_s_t*)(device->pad))[port];
-	int32_t rtn = ((vexDeviceAdiValueGet(device->device_info, port) << 4) - adi_data->analog_data.calib);
-	return_port(INTERNAL_ADI_PORT, rtn);
+	return ext_adi_analog_read_calibrated_HR(port, INTERNAL_ADI_PORT);
 }
 
 int32_t adi_digital_read(uint8_t port) {
-	transform_adi_port(port);
-	claim_port_i(INTERNAL_ADI_PORT, E_DEVICE_ADI);
-	validate_type(device, port, E_ADI_DIGITAL_IN);
-	int32_t rtn = vexDeviceAdiValueGet(device->device_info, port);
-	return_port(INTERNAL_ADI_PORT, rtn);
+	return ext_adi_digital_read(port, INTERNAL_ADI_PORT);
 }
 
 int32_t adi_digital_get_new_press(uint8_t port) {
-	transform_adi_port(port);
-	claim_port_i(INTERNAL_ADI_PORT, E_DEVICE_ADI);
-	validate_type(device, port, E_ADI_DIGITAL_IN);
-
-	adi_data_s_t* const adi_data = &((adi_data_s_t*)(device->pad))[port];
-
-	int32_t pressed = vexDeviceAdiValueGet(device->device_info, port);
-
-	if (!pressed)
-		adi_data->digital_data.was_pressed = false;
-	else if (!adi_data->digital_data.was_pressed) {
-		// Button is currently pressed and was not detected as being pressed during last check
-		adi_data->digital_data.was_pressed = true;
-		return_port(INTERNAL_ADI_PORT, true);
-	}
-
-	return_port(INTERNAL_ADI_PORT, false);
+	return ext_adi_digital_get_new_press(port, INTERNAL_ADI_PORT);
 }
 
 int32_t adi_digital_write(uint8_t port, const bool value) {
```

Comment:
Why is a non-reference bool marked as const?
Same with `ext_adi.h:L418` and others.